### ISSUE TITLE
Add support for constrasts

### ISSUE THREAD

**Issue by pitakakariki** [pitakakariki/simr#76]
Currently `extend` drops contrasts, which tends to cause errors because the factors have different names.

**Comment by pitakakariki** [comment]
Shouldn't fall over now, as long as no new levels are added to a factor with contrasts.

**Comment by behinger** [comment]
Two questions:
1. Are contrasts stable now? I saw a "still experimental". Where are the caveats / what is out there to look for?

2. How would I specify contrasts? E.g. I have two Factors A,B and I want the contrast c(1,-1).

**Comment by pitakakariki** [comment]
"Still experimental" might be a bit over-cautious. I've fixed the bug that was breaking powerSim when it was given a model with contrasts, and as far as I can tell it's working now - at least no one has reported any bugs from using it.

I do still have to write some unit tests for it before I'll be confident enough to call it "stable" and push it to CRAN.

I don't use contrasts much myself (which is why I didn't originally add support for them in simr) but I think this page should tell you how to set the contrasts you want: https://www.rdocumentation.org/packages/stats/versions/3.4.1/topics/contrasts

**Comment by behinger** [comment]
ah I see, I thought of contrasts as a more general concept that allows more complicated things not specific to a single factor. 
Thanks!

**Comment by sollago** [comment]
Sorry for repeating the question above, but I just installed `simr` and I think that running simulations with manually specified contrasts is still not working: in my case, the problem does not occur with `extend` but with `powerSim` more generally. The contrasts were defined as  `sumcoding = -contr.sum(2)/2 ` and the model: `m <- lmer(log(value) ~ factor1*factor2 + (1|participant), contrasts =list(factor1=sumcoding, factor2=sumcoding), data=data)`.  Then, when I run `powerSim(m, fixed('factor1:factor2', 'z'))` I still get an output that says that out of 1000 simulations, 1000 yielded errors. When I remove the contrasts from the model call, the problem does no longer occur. Let me know if you have any advice and thanks for putting the package together (it is really useful)!

**Comment by pitakakariki** [comment]
Which version of simr are you using? The contrasts fix is only in the github version at this point, and hasn't made it to cran yet.

**Comment by sollago** [comment]
Ah, that makes sense: I am using the latest CRAN version (simr_1.0.3). Thanks, and looking forward to the CRAN update!

**Comment by liznecka** [comment]
I updated to the version from github, but when I run an analysis with custom contrasts, I get the following errors in lastResults()$errors:
random effects specified in re.form that were not present in original model

Any idea on why this would be happening? I've stripped all NAs from my data frame.  I get the same error regardless of whether I list the specific fixed effects (e.g., with the contrast levels) or the more general fixed effect in my test = fixed() option.  I can send you a reproducible example via email if necessary.
